File: benchmarks/tsb/bench_series_filter.ts
/**
 * Benchmark: Series.filter(mask) — boolean selection on 100k Series.
 */
import { Series } from "../../src/index.js";

const SIZE = 100_000;
const WARMUP = 5;
const ITERATIONS = 50;

const s = new Series({ data: Array.from({ length: SIZE }, (_, i) => i) });
const mask = new Series({ data: Array.from({ length: SIZE }, (_, i) => i % 2 === 0) });

for (let i = 0; i < WARMUP; i++) s.filter(mask);

const times: number[] = [];
for (let i = 0; i < ITERATIONS; i++) {
  const t0 = performance.now();
  s.filter(mask);
  times.push(performance.now() - t0);
}
const total = times.reduce((a, b) => a + b, 0);
console.log(JSON.stringify({ function: "series_filter", mean_ms: total / ITERATIONS, iterations: ITERATIONS, total_ms: total }));

File: benchmarks/pandas/bench_series_filter.py
"""Benchmark: Series boolean selection on 100k Series."""
import json, time
import numpy as np
import pandas as pd

SIZE = 100_000
WARMUP = 5
ITERATIONS = 50

s = pd.Series(np.arange(SIZE))
mask = pd.Series(np.arange(SIZE) % 2 == 0)
for _ in range(WARMUP): s[mask]

times = []
for _ in range(ITERATIONS):
    t0 = time.perf_counter()
    s[mask]
    times.append(time.perf_counter() - t0)
total = sum(times) * 1000
print(json.dumps({ "function": "series_filter", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total }))
